File: quests/qeynos/Exterminator_Rasmon.pl
sub EVENT_SAY {
  if($text=~/Hail/i){ #Start of Exterminate the Vermin Quest (Scripted By: Fatty Beerbelly)
    quest::say("Hello, $name. Have you seen these pesky rodents? They are everywhere, and I simply cannot stand them! If you are willing to do a little cleaning up of the pests here, I will reward you for every four rat whiskers you bring me.");
  }
}

sub EVENT_ITEM {
  #Handin: 4x Rat Whiskers
  if(plugin::check_handin(\%itemcount, 13071 => 4)){ #End of Exterminate the Vermin (Scripted By: Fatty Beerbelly)
    quest::say("I am very impressed, $name! A few more cleaners like yourself and we could have a rodent-free Qeynos in no time!");
    quest::faction(135,10); # +Guards of Qeynos
    quest::faction(9,10);   # +Antonius Bayle
    quest::faction(217,10); # +Merchants of Qeynos
    quest::faction(53,-10); # -Corrupt Qeynos Guards
    quest::faction(33,-10); # -Circle of Unseen Hands
    quest::exp(5);
    quest::givecash(1,1,1,0);
  }
}

File: quests/qeynos/Exterminator_Rasmon.lua
function event_say(e)
	if(e.message:findi("hail")) then
		e.self:Say(string.format("Hello, %s. Have you seen these pesky rodents? They are everywhere, and I simply cannot stand them! If you are willing to do a little cleaning up of the pests here, I will reward you for every four rat whiskers you bring me.",e.other:GetName()));
	end
end

function event_trade(e)
	local item_lib =require("items");
	if(item_lib.check_turn_in(e.trade, {item1 = 13071,item2 = 13071,item3 = 13071,item4 = 13071})) then
		e.self:Say(string.format("I am very impressed, %s! A few more cleaners like yourself and we could have a rodent-free Qeynos in no time!",e.other:GetName()));
		e.other:Ding();
		e.other:Faction(135,5,0);
		e.other:Faction(9,5,0);
		e.other:Faction(53,-5,0);
		e.other:Faction(33,-5,0);
		e.other:Faction(217,5,0);
		e.other:AddEXP(50);
		e.other:GiveCash(0,4,0,0);
	end
	item_lib.return_items(e.self, e.other, e.trade)
end
